pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader.go

```diff
@@ -48,12 +48,12 @@ func EncodeVersionedBytes(data any, version uint32) (*pb.VersionedBytes, error)
 
 	switch version {
 	case JSONEncodingVersion1:
-		bytes, err = jsonv1.Marshal(data)
+		bytes, err = json.Marshal(data)
 		if err != nil {
 			return nil, fmt.Errorf("%w: %w", types.ErrInvalidType, err)
 		}
 	case JSONEncodingVersion2:
-		bytes, err = jsonv2.Marshal(data)
+		bytes, err = json.Marshal(data)
 		if err != nil {
 			return nil, fmt.Errorf("%w: %w", types.ErrInvalidType, err)
 		}
```

Comment:
This was using jsonV2 now it's using jsonV1.

To be fair, it seems odd that they have different encoding/decoding versions because only go would know the difference...

Reply:
I don't know on this one. I was considering Jordan's feedback on what we're considering `v2` for json.

Reply:
In this context, we have an explicit switch for the two different versions, so why wouldn't we use v2 here inside of the v2 case?

Reply:
> To be fair, it seems odd that they have different encoding/decoding versions because only go would know the difference...

Indeed, these should be renamed from `Encoding` to `Encoder`

Reply:
I switched it back and on reviewing the docs, there was an option to stringify numbers, which kinda made sense with what we were working with so I added that.

Reply:
Does that necessarily work the same way?
I think we'd need something like RawValue: https://pkg.go.dev/github.com/go-json-experiment/json@v0.0.0-20240524174822-2d9f40f7385b#example-WithUnmarshalers-RawNumber
Because even with `StringifyNumbers`, unmarshaling without a particular target type in mind results in `float64` values.

Reply:
The float/int problem wasn't exposed in the grpc encoding/decoding. It was in the `ChainReader` config encoding/decoding due to `map[string]any` being used as types to decode into.

ex.
```
{"hardCodeProp": 3} --> decodes to map[string]any{"hardCodeProp": float64(3)}
```

Then the hard coder adds a property to a type:
```
type A struct {
    B string
}

type NewType struct {
    B string
    NewProp float64
}
```

When `GetLatestValue` is called with the following type, the error would present itself where the `NewProp` is being added by the hard-coder. The expected type is an 'int', but the hard-coder thinks it should be a `float64` because of the config decoding.
```
type Expected struct {
    B string
    NewProp int
}
```

Reply:
To be fair, anytime I switched the grpc encoding/decoding to json, I ran into more errors on when running the evm tests. If we intend to use this in the future, I think we will need to explore a lot more than just numbers.

Reply:
I'm having trouble following this scenario. But if we don't have active test coverage on these encodings, then IMHO we shouldn't be messing with them until that is added.